### DISCUSSION TITLE
Arrow Date format

### DISCUSSION THREAD

**Discussion by ryanmckaytx** [observablehq/framework#1601] [Q&A] [answered]
I'm using fenced sql to query a local parquet file of generated credit card transaction data.  There are columns `amt` and `trans_date` in the data.  Using 
````
```sql id=summarized_txn
summarize txn1mm
```
```js
view(Inputs.table(summarized_txn))
```
````
yields
| column_name | column_type | min                | max |
| --- | --- | --- | --- |
| trans_date | DATE | 2023-01-01 | 2023-12-31 |
| amt | DOUBLE | 1.0 | 26810.71 |

but viewing the actual records of the dataset in a table gives values like `1,703,894,400,000` for `trans_date` (i.e. unix millisecond timestamp).  

I'm trying to generate a bar chart with sum of `amt` as the bar height and `trans_date` as the x value.  The binning/summing is working as expected, but the x values look like the int value, not a date.  The chart is specified as:
````
```js
function salesBars(data, {width} = {}) {
    return Plot.plot({
        marks: [
            Plot.barY(data, Plot.binX({y: "sum"}, {x:(d) => d.trans_date, y:"amt", tip:true}))
        ],
        width: width,
    });
};
```
````
I tried treating creating Date objects out of trans_date values, but that results in a single bar with the sum of all the values
````
```js
function salesBars(data, {width} = {}) {
    return Plot.plot({
        marks: [
            Plot.barY(data, Plot.binX({y: "sum"}, {x:(d) => new Date(d.trans_date).toISOString(), y:"amt", tip:true}))
        ],
        width: width,
    });
};
```
````
any ideas how to get x values in yyyy-mm-dd format?  Sorry if the answer is obvious, I'm a JS and observable novice.

**Comment by mbostock** [comment]
What version of Framework, Inputs, and Plot are you using? We recently made improvements to our Apache Arrow support to accommodate Arrow’s use of BigInt to represent dates. You might try upgrading your dependencies, and see the latest release notes for these libraries.

https://github.com/observablehq/framework/issues/1376

**Reply by ryanmckaytx** [reply]
npm ls outputs
```
├── @observablehq/framework@1.10.1
├── d3-dsv@3.0.1
├── d3-time-format@4.1.0
└── rimraf@5.0.10
```
I think part of my problem is not fully groking bar vs rect marks.  I played around with a few approaches and rectY treated the dates as dates with no problem.  I had bar chart in my head so I went straight to bar.  Looking at the Plot examples helped steer me to rect. 
 The different attempts are here: https://ryanmckaytx.github.io/observable-framework-demo/bar-chart-tests

**Reply by mbostock** [reply]
Inputs is a dependency of Framework, and I don’t see that in your output so I can’t tell what version of Inputs you are using. For Plot, you’ll want to clear your npm cache which is typically in `src/.observablehq/cache/_npm`.

**Reply by mbostock** [reply, marked as answer]
Looking at the live example you shared (thank you for that) it appears you are using the latest versions of Framework (1.10.1), Plot (0.6.16), and Inputs (0.11.0). So you should be all good there.

For bar _vs._ rect marks, you can use either depending on whether you want to treat dates as ordinal or quantitative. In either case, you typically want to specify the [**interval** option](https://observablehq.com/plot/transforms/interval) to tell Plot what the frequency of the samples are, such as `interval: "day"` for daily data. And you generally use the barY mark with the groupX transform, and the rectY mark with the binX transform.  Using barY with binX is generally a no-no.

If you have additional questions about Plot, please ask over on the [Plot discussions](https://github.com/observablehq/plot/discussions) for the benefit of other Plot users. Thank you!
